api/mappers.go

```diff
@@ -214,9 +215,180 @@ func NewPatientDto(patient *patients.Patient) PatientV1 {
 		}}
 	}
 
+	if patient.DiagnosisType != "" {
+		dto.DiagnosisType = NewDiagnosisTypeDto(patient.DiagnosisType)
+	}
+
 	return dto
 }
 
+func NewDiagnosisTypeDto(diagnosisType string) *DiagnosisTypeV1 {
+	dtoDiagnosisType := DiagnosisTypeV1(diagnosisType)
+	switch dtoDiagnosisType {
+	case DiagnosisTypeV1Gestational:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Lada:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Mody:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1NotApplicable:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Other:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Prediabetes:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Type1:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Type2:
+		return &dtoDiagnosisType
+	case DiagnosisTypeV1Type3c:
+		return &dtoDiagnosisType
+	default:
+		return nil
+	}
+}
+
+func NewGlycemicRangesDto(ranges patients.GlycemicRanges) *GlycemicRangesV1 {
+	v1Ranges, err := newGlycemicRangesDtoWithError(ranges)
+	if err != nil {
+		return nil
+	}
+	return v1Ranges
+}
+
+func newGlycemicRangesDtoWithError(ranges patients.GlycemicRanges) (
+	*GlycemicRangesV1, error) {
+
+	switch ranges.Type {
+	case "custom":
+		custom, err := newGlycemicRangesDtoCustom(ranges.Custom)
+		if err != nil {
+			return nil, err
+		}
+		return &GlycemicRangesV1{Type: "custom", Custom: *custom}, nil
```

Comment:
Same here `patients.GlycemicRangeTypeCustom`

Reply:
This has to be `Custom` (found here in api), but sure, I can make that change.